lib/applications/routes.js

```diff
@@ -4,53 +4,317 @@
  * @fileoverview Applications routing functions
  *
  */
+const config = require('config');
+const _ = require('lodash');
+const types = require('punchcard-content-types');
+const uuid = require('uuid');
+
+const database = require('../database');
+const model = require('./model');
+const utils = require('../utils');
 
 /**
- * All Webhooks Landing page
+ * All Applications Landing page
  * @param {object} req - HTTP Request
  * @param {object} res - HTTP Response
  * @param {object} next - Express callback
  *
+
+ * @returns {promise} promise wrapping database promise
  */
-const all = () => {};
+const all = (req, res, next) => {
+  return database
+    .select('*')
+    .from('applications')
+    .orderBy('name', 'DESC').then(rows => {
+      const applications = rows.map(rw => {
+        const row = rw;
+        row.client = {
+          id: row['client-id'],
+          secret: row['client-secret'],
+        };
+
+        return row;
+      });
+
+      res.render('applications/all', {
+        applications,
+        config: config.applications,
+        structure: model.structure,
+      });
+
+      return true;
+    }).catch(e => {
+      const err = {
+        message: e.message,
+        safe: '/',
+        status: 500,
+      };
+
+      return next(err);
+    });
+};
 
 /**
- * Add a new Webhook
+ * Add a new Application
  * @param {object} req - HTTP Request
  * @param {object} res - HTTP Response
  * @param {object} next - Express callback
  *
+ * @returns {promise} content-types promise
  *
  * TODO: Content-Types errors need to be vastly expanded to properly use promise catch for error
  * https://github.com/punchcard-cms/content-types/issues/112
  */
-const add = () => {};
+const add = (req, res, next) => {
+  const errors = _.get(req.session, 'form.applications.save.errors', {});
+  const values = _.get(req.session, 'form.applications.save.content', {});
+
+  _.unset(req.session, 'form.applications.save');
+
+  return types.only('applications', values, [req.applications.merged], config).then(merged => {
+    return types.form(merged, errors, config).then(form => {
+      res.render('applications/add', {
+        form,
+        action: req.url.replace(config.applications.actions.add, config.applications.actions.save),
```

Comment:
This is what got us in to trouble with the content routes. Can we take the same tactic we did there for this action?